charts/consul/templates/api-gateway-gatewayclassconfig.yaml

```diff
@@ -0,0 +1,47 @@
+{{- if (or (and (ne (.Values.apiGateway.enabled | toString) "-") .Values.apiGateway.enabled) (and (eq (.Values.apiGateway.enabled | toString) "-") .Values.global.enabled)) }}
+{{- if .Values.apiGateway.managedGatewayClass.enabled }}
+apiVersion: api-gateway.consul.hashicorp.com/v1alpha1
+kind: GatewayClassConfig
+metadata:
+  name: consul-api-gateway-class-config
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: api-gateway
+spec:
+  consul:
+    authentication:
+      {{- if .Values.global.acls.manageSystemACLs }}
+      managed: true
+      method: {{ template "consul.fullname" . }}-k8s-auth-method
```

Comment:
This is read by the api gateway controller. Managed means that the acl system is managed so we create a service account for each logical gateway that matches the name we register the service as so that each gateway instance can auth to consul via the configured ACL auth method. By default managed is false and the controller expects an `account` to be set to the name of a operator managed ServiceAccount.

Reply:
Awesome, thanks for the explanation!